opengrok-indexer/src/main/java/org/opengrok/indexer/history/MercurialRepository.java

```diff
@@ -104,6 +105,12 @@
     private static final Pattern LOG_COPIES_PATTERN
             = Pattern.compile("^(\\d+):(.*)");
 
+    /**
+     * This is a static replacement for 'working' field. Effectively, check if hg is working once in a JVM
+     * instead of calling it for every MercurialRepository instance.
+     */
+    private static final LazilyInstantiate<Boolean> HG_IS_WORKING = LazilyInstantiate.using(MercurialRepository::isHgWorking);
```

Comment:
Just a note for future, I'd personally change the name of `LazilyInstantiate` to `MemoizingSupplier` which follows more common naming or even better use one from Guava https://guava.dev/releases/19.0/api/docs/com/google/common/base/Suppliers.html#memoize(com.google.common.base.Supplier)

Reply:
Also, I think using supplier type is nicer:
 `Supplier<Boolean> HG_IS_WORKING = ...`

Reply:
ok, used Supplier.